include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp

```diff
@@ -501,16 +497,12 @@ struct GemmKernel
         // Run Epilogue Pipeline
         auto& c_block_window = gemm_tile_windows.at(I2);
 
-        constexpr bool is_output_c_reg_transposed =
-            EpiloguePipeline::IsOutputTransposed() != GemmPipeline::IsTransposeC();
-        if constexpr((DstInMemOp == memory_operation_enum::set) || (sizeof(CDataType) > 2) ||
-                     (GemmPipeline::VectorSizeC % 2 == 0 &&
-                      std::is_same_v<CLayout, tensor_layout::gemm::RowMajor> &&
-                      is_output_c_reg_transposed))
+        if constexpr(DstInMemOp == memory_operation_enum::set || sizeof(CDataType) > 2 ||
+                     EpiloguePipeline::GetVectorSizeC() % 2 == 0)
         {
             EpiloguePipeline{}
                 .template operator()<decltype(c_block_window), decltype(c_block_tile), DstInMemOp>(
-                    c_block_window, c_block_tile);
+                    c_block_window, c_block_tile, smem_ptr);
         }
     }
```

Comment:
This looks strange. Like we run Epilogue only under specific conditions...

Reply:
We need this because if split K is not suppoted due to vector size then algorithm could be supported for split_k=1. If we remove this condition then it will not compile. IsSupported should not run this kernel in case of split_k not supported